src/library.cpp

```diff
@@ -123,6 +125,64 @@ void Library::save()
     m_library.writeBookmarksToFile(kiwix::appendToDirectory(m_libraryDirectory.toStdString(), "library.bookmarks.xml"));
 }
 
+void Library::setMonitorDirZims(QStringList zimList)
+{
+    m_monitorDirZims = zimList;
+}
+
+QStringList Library::getLibraryZimsFromDir(QString dir)
+{
+    QStringList zimsInDir;
+    for (auto str : getBookIds()) {
+        auto filePath = QString::fromStdString(getBookById(str).getPath());
+        QDir absoluteDir = QFileInfo(filePath).absoluteDir();
+        if (absoluteDir == dir) {
+            zimsInDir.push_back(filePath);
+        }
+    }
+    return zimsInDir;
+}
+
+void Library::loadMonitorDir(QString monitorDir)
+{
+    QMutex mutex;
+    QMutexLocker locker(&mutex);
+    const QDir dir(monitorDir);
+    QStringList newDirEntries = dir.entryList({"*.zim"});
+    for (auto &str : newDirEntries) {
+        str = monitorDir + QDir::separator() + str;
+    }
+    QSet<QString> newDir = QSet<QString>::fromList(newDirEntries);
+    QStringList oldDirEntries = getMonitorDirZims();
+    QSet<QString> oldDir = QSet<QString>::fromList(oldDirEntries);
+    QSet<QString> addedZimSet = newDir - oldDir;
+    QStringList addedZims = addedZimSet.values();
+    QSet<QString> removedZimSet = oldDir - newDir;
+    QStringList removedZims = removedZimSet.values();
```

Comment:
Eliminate the intermediate variable `removedZimSet` by combining these two lines into one.

Reply:
This comment was not addressed (or the commit was lost)

Reply:
This comment is still not addressed